numpy/lib/tests/test_io.py

```diff
@@ -1272,6 +1272,15 @@ def test_commented_header(self):
             assert_(w[0].category is np.VisibleDeprecationWarning)
         assert_equal(test, ctrl)
 
+    def test_names_and_comments_none(self):
+        #tests case when names is true but comments is None
+        data = TextIO('col1 col2\n 1 2\n 3 4')
+        with warnings.catch_warnings(record=True) as w:
+            warnings.filterwarnings('always', '', np.VisibleDeprecationWarning)
```

Comment:
What's going on here?

Reply:
To catch warnings...if it's not necessary I'll remove those 2 lines

Reply:
What warnings are you producing?

Reply:
This code is present in npyio.py